celery/beat.py

```diff
@@ -257,7 +258,9 @@ def tick(self, event_t=event_t, min=min, heappop=heapq.heappop,
         adjust = self.adjust
         max_interval = self.max_interval
 
-        if self._heap is None:
+        if (self._heap is None or
+                not self.schedules_equal(self.old_schedulers, self.schedule)):
```

Comment:
The [schedule](https://github.com/celery/celery/blob/master/celery/schedules.py#L171) class has defined the `_eq__` method, so I think you can compare directly the dictionaries `self.old_schedulers` and `self.schedule`, you should not need an additional function to achieve this.